pkg/services/datasourceproxy/datasourceproxy.go

```diff
@@ -53,30 +54,54 @@ type DataSourceProxyService struct {
 func (p *DataSourceProxyService) ProxyDataSourceRequest(c *models.ReqContext) {
 	id, err := strconv.ParseInt(web.Params(c.Req)[":id"], 10, 64)
 	if err != nil {
-		c.JsonApiErr(http.StatusBadRequest, "id is invalid", err)
+		p.ProxyDatasourceRequestWithUID(c)
```

Comment:
What if the UID is numeric, but still not same as the ID. Tricky one :) The only additional alternative I would add is introducing a new specific UID route.

Reply:
> What if the UID is numeric, but still not same as the ID. Tricky one :)
that's a good point!

Another route is an option but I have to use a slightly different path so that requests are routed properly there. The other option would be to require a prefix for the datasource UID path parameter: like `uid:PDDA8E780A17E7EF1`
WDYT?

Reply:
thinking more about this I think I prefer the additional route. it's clearer

Reply:
Hmm tricky.

> but I have to use a slightly different path so that requests are routed properly there

You mean that you'll have to change other code to use a different path? 

> The other option would be to require a prefix for the datasource UID path parameter: like `uid:PDDA8E780A17E7EF1`

That would be same work as my suggestion introducing a new route I guess so then I would favor adding a new route. 

Is the idea to remove support for sequential ID's in v9?

Reply:
> Is the idea to remove support for sequential ID's in v9?

I think the idea is to deprecate them by v9 but not remove them yet. I'm not sure about keep supporting them; I guess we will still do until they will be eventually removed.